frontend/src/tests/UsernameInput.spec.ts

```diff
@@ -0,0 +1,194 @@
+                userId: 'userId',
+            },
+            status: 200,
+            statusText: 'Ok',
+            headers: {},
+            config: {
+                headers: {} as AxiosRequestHeaders,
+            },
+        };
+    } else {
+        return {
+            data: {
+                found: false,
+                userId: 'userId',
+            },
+            status: 200,
+            statusText: 'Ok',
+            headers: {},
+            config: {
+                headers: {} as AxiosRequestHeaders,
+            },
+        };
+    }
+};
+
+describe('UserNameInput', () => {
+    let wrapper: VueWrapper;
+    let emitSetVerifyResult: unknown[][] | undefined;
+
+    let usernameInputText: DOMWrapper<HTMLElement>;
+    let usernameInputTextEl: HTMLInputElement;
+    let verifyButton: DOMWrapper<HTMLElement>;
+    let verifyButtonEl: HTMLButtonElement;
+    let userIdentity;
+
+    const props = {
+        domain: UserType.I,
+        userId: '',
+        fieldId: 'userId',
+    };
+
+    const newProps = {
+        domain: UserType.B,
+        userId: 'newUserId',
+        fieldId: 'newFieldId',
+    };
+
+    const newValue = 'testIdir';
+
+    beforeEach(async () => {
+        wrapper = mount(UserNameInput, {
+            props,
+        });
+
+        usernameInputText = wrapper.find('#userIdInput');
+        usernameInputTextEl = usernameInputText.element as HTMLInputElement;
+        verifyButton = wrapper.find("[data-target-btn='verifyIdir']");
+        verifyButtonEl = verifyButton.element as HTMLButtonElement;
+    });
+
+    afterEach(() => {
+        vi.clearAllMocks();
+        wrapper.unmount();
+    });
+
+    it('Should change usernameInput value', async () => {
+        expect(usernameInputTextEl.value).toBe('');
+        await usernameInputText.setValue(newValue);
+        expect(usernameInputTextEl.value).toBe(newValue);
+    });
+
+    it('Should receive the correct prop', async () => {
+        //default props
+        expect(wrapper.props()).toEqual(props);
+
+        await wrapper.setProps(newProps);
+        expect(wrapper.props()).toEqual(newProps);
+        expect(wrapper.props()).not.toEqual(props);
+    });
+
+    it('Should call and emit correct value', async () => {
+        await usernameInputText.setValue(newValue);
+        const emitChange = wrapper.emitted('change');
+        expect(emitChange).toBeTruthy();
+        // test the given parameters when emitChange has been called
+        // i.e. emitChange = [ [ 'B' ] ]
+        expect(emitChange![0][0]).toEqual(newValue);
+
+        emitSetVerifyResult = wrapper.emitted('setVerifyResult');
+        await wrapper.setProps(newProps);
+        await verifyButton.trigger('click');
+
+        expect(emitSetVerifyResult).toBeTruthy();
+        expect(emitSetVerifyResult![1][0]).toEqual(true);
+    });
+
+    it('Should enable verify btn when username is inputted', async () => {
+        // button starts as disabled
```

Comment:
here we need to expect the input is empty initially

Reply:
sure, will do